ts/components/conversation/ContactDetail.tsx

```diff
@@ -26,47 +26,74 @@ interface Props {
   onSendMessage: () => void;
 }
 
-function getLabelForContactMethod(method: Phone | Email, i18n: Localizer) {
+function getLabelForEmail(method: Email, i18n: Localizer): string {
   switch (method.type) {
     case ContactType.CUSTOM:
-      return method.label;
+      return method.label || i18n('email');
     case ContactType.HOME:
       return i18n('home');
     case ContactType.MOBILE:
       return i18n('mobile');
     case ContactType.WORK:
       return i18n('work');
     default:
-      return missingCaseError(method.type);
+      throw missingCaseError(method.type);
   }
 }
 
-function getLabelForAddress(address: PostalAddress, i18n: Localizer) {
+function getLabelForPhone(method: Email, i18n: Localizer): string {
```

Comment:
Should `method` be `Phone` here? It might be possible to generalize this function to work for both email and phone but I am cool keeping it as-is.